src/components/tabset/tabset.component.spec.ts

```diff
@@ -0,0 +1,114 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { TabsetModule } from './tabset.module';
+import { Component } from '@angular/core';
+import { dispatchMouseEvent } from '../../common/testing';
+
+@Component({
+    selector: 'app-radio-button-value-test',
+    template: `
+        <ux-tabset [minimal]="true" aria-label="Tabset Example">
+            <ux-tab (deselect)="onTabDeselect()"  (deactivated)="onTabDeactivated()" (select)="onTabSelectClick()"  (activated)="onTabActivatedClick()" customClass="text-center" *ngFor="let tab of tabs">
+                <ng-template uxTabHeading>
+                    <span>{{ tab.title }}</span>
+                </ng-template>
+
+                <div class="m-t">
+                    <h4>{{ tab.title }}</h4>
+                    <p>{{ tab.content }}</p>
+                </div>
+            </ux-tab>
+        </ux-tabset>
+    `
+})
+export class TabsetTestComponent {
+
+    tabs: Tab[] = [
+        {
+            content: 'test text',
+            title: 'Schedule',
+        },
+        {
+            content: 'test text',
+            title: 'Protection',
+        },
+        {
+            content: 'test text',
+            title: 'Solution',
+        },
+        {
+            content: 'test text',
+            title: 'Analytics',
+        }
+    ];
+
+    onTabSelectClick(): void {
+    }
+
+    onTabActivatedClick(): void {
+    }
+
+    onTabDeselect(): void {
+    }
+
+    onTabDeactivated(): void {
+    }
+}
+
+describe('Radio Button Component - Value', () => {
+    let component: TabsetTestComponent;
+    let fixture: ComponentFixture<TabsetTestComponent>;
+    let nativeElement: HTMLElement;
+
+    beforeEach(async () => {
+        await TestBed.configureTestingModule({
+            imports: [TabsetModule],
+            declarations: [TabsetTestComponent]
+        }).compileComponents();
+
+        fixture = TestBed.createComponent(TabsetTestComponent);
+        component = fixture.componentInstance;
+        nativeElement = fixture.nativeElement;
+        fixture.detectChanges();
+    });
+
+    it('should initialise correctly', () => {
+        expect(component).toBeTruthy();
+    });
+
+    it('should emit activated output when tab is clicked and emit deactivated output when tab deselected', () => {
```

Comment:
Could we split the `activated` and `deactivated` tests into separate tests, this way if a failure occurs we will know which one is not working (or if both are not working)

Reply:
tests separated